platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/resolver/DefaultResolutionOutputs.java

```diff
@@ -0,0 +1,311 @@
+            this.reselectVariants = reselectVariants;
+            this.viewAttributes = viewAttributes;
+
+            this.resultProvider = resultProvider;
+            this.attributesFactory = attributesFactory;
+            this.calculatedValueContainerFactory = calculatedValueContainerFactory;
+            this.resolutionHost = resolutionHost;
+            this.taskDependencyFactory = taskDependencyFactory;
+        }
+
+        @Override
+        public ArtifactCollectionInternal getArtifacts() {
+            return new DefaultArtifactCollection(
+                getFiles(),
+                lenient,
+                resolutionHost,
+                calculatedValueContainerFactory
+            );
+        }
+
+        @Override
+        public ResolutionBackedFileCollection getFiles() {
+            return new ResolutionBackedFileCollection(
+                resultProvider.map(this::selectArtifacts),
+                lenient,
+                resolutionHost,
+                taskDependencyFactory
+            );
+        }
+
+        private SelectedArtifactSet selectArtifacts(ResolverResults results) {
+            // If the user set the view attributes, we allow variant matching to fail for no matching variants.
+            // If we are using the original request attributes, variant matching should not fail.
+            boolean allowNoMatchingVariants = !viewAttributes.isEmpty();
+
+            return results.getVisitedArtifacts().select(new ArtifactSelectionSpec(
+                computeViewAttributes(results),
+                componentFilter,
+                reselectVariants,
+                allowNoMatchingVariants,
+                results.getVisitedArtifacts().getDefaultSortOrder()
+            ));
+        }
+
+        private ImmutableAttributes computeViewAttributes(ResolverResults results) {
+            ImmutableAttributes baseAttributes = results.getVisitedGraph().getResolutionResult().getRequestedAttributes();
+
+            // The user did not specify any attributes. Use the original request attributes.
+            if (viewAttributes.isEmpty()) {
+                return baseAttributes;
+            }
+
+            // When re-selecting, we do not base the view attributes on the original request attributes.
+            if (reselectVariants) {
+                return viewAttributes;
+            }
+
+            // Otherwise, artifact views without re-selection are based on the original request attributes.
+            return attributesFactory.concat(baseAttributes, viewAttributes);
+        }
+
+        @Override
+        @Deprecated
+        public ImmutableAttributes getAttributes() {
+            DeprecationLogger.deprecateMethod(ArtifactView.class, "getAttributes()")
+                .willBecomeAnErrorInGradle9()
+                .withUpgradeGuideSection(8, "deprecate_artifact_view_get_attributes")
+                .nagUser();
+
+            return computeViewAttributes(resultProvider.getValue());
```

Comment:
Its weird that calling this method causes resolution. 
Can we temporarily wire these input attributes to this class until the deprecation happens?